src/huggingface_hub/hf_api.py

```diff
@@ -1572,6 +1676,14 @@ def list_datasets(
         limit: Optional[int] = None,
         full: Optional[bool] = None,
         token: Optional[str] = None,
+        benchmark: Optional[Union[str, List[str]]] = None,
+        dataset_name: Optional[str] = None,
+        language_creators: Optional[Union[str, List[str]]] = None,
+        language: Optional[Union[str, List[str]]] = None,
+        multilinguality: Optional[Union[str, List[str]]] = None,
+        size_categories: Optional[Union[str, List[str]]] = None,
+        task_categories: Optional[Union[str, List[str]]] = None,
+        task_ids: Optional[Union[str, List[str]]] = None,
```

Comment:
Same comment as for `list_models` (would prefer to have them between `author` and `search`).